### PR TITLE
Make it possible to register a own FuncLayoutRenderer

### PR DESCRIPTION
**Pull request by 304NotModified**
Needed for https://github.com/NLog/NLog.Web/issues/632

### CODE REVIEW

File: src/NLog/LayoutRenderers/FuncLayoutRenderer.cs [hunk 1]
```diff
@@ -92,9 +104,14 @@ private string GetStringValue(LogEventInfo logEvent)
             return null;
         }
 
-        private object GetValue(LogEventInfo logEvent)
+        /// <summary>
+        /// Get the value for the LogEvent
+        /// </summary>
+        /// <param name="logEvent"></param>
+        /// <returns></returns>
+        protected virtual object GetValue(LogEventInfo logEvent)
```
Comment (snakefoot) [inline src/NLog/LayoutRenderers/FuncLayoutRenderer.cs:112]: Please rename to `RenderValue`. Since it is a LayoutRenderer.

Comment (304NotModified) [reply src/NLog/LayoutRenderers/FuncLayoutRenderer.cs]: done

File: src/NLog/LayoutRenderers/FuncLayoutRenderer.cs [hunk 2]
```diff
@@ -85,16 +97,21 @@ private string GetStringValue(LogEventInfo logEvent)
...
+                object value = RenderValue(logEvent);
                 string stringValue = FormatHelper.TryFormatToString(value, Format, GetFormatProvider(logEvent, null));
                 return stringValue;
             }
             return null;
         }
 
-        private object GetValue(LogEventInfo logEvent)
+        /// <summary>
+        /// Get the value for the LogEvent
```
Comment (snakefoot) [inline src/NLog/LayoutRenderers/FuncLayoutRenderer.cs:108]: Render the value from the LogEvent

Comment (304NotModified) [reply src/NLog/LayoutRenderers/FuncLayoutRenderer.cs]: rewrote it.

**Review by snakefoot — APPROVED** [review]

**Pull request merged**